### PR TITLE
Loader refactor

### PR DESCRIPTION
**Pull request by dwoz**
### What does this PR do?

- Refactor loader into smaller, more logical files.
- Add better docs LoaderContext changes
- Add changelog for LoaderContext changes

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### CODE REVIEW

**Review by Akm0d — DISMISSED** [review]

File: doc/topics/development/modules/developing.rst [hunk 1]
```diff
@@ -61,6 +61,44 @@ supports tab-completion.
 
 To exit IPython and continue running Salt, press ``Ctrl-d`` to logout.
 
+.. _loader:
+
+The Salt Loader
+===============
+
+Salt's loader system is resposible for reading `Special Module Contents`_ and
```
Comment (eliasp) [inline doc/topics/development/modules/developing.rst:69]:
```suggestion
Salt's loader system is responsible for reading `Special Module Contents`_ and
```

**Review by s0undt3ch — COMMENTED** [review]
I would take this a step further.
`salt.loader.__init__.py` would only contain imports, likely just the loader functions imports.
Any support/helper function that is used in more than one submodule would go into `salt.loader.helpers` or `salt.loader.utils`.
All loader functions would go into `salt.loader.loaders`.

Thoughts?

**Comment by dwoz** [comment]
> I would take this a step further.
> `salt.loader.__init__.py` would only contain imports, likely just the loader functions imports.
> Any support/helper function that is used in more than one submodule would go into `salt.loader.helpers` or `salt.loader.utils`.
> All loader functions would go into `salt.loader.loaders`.
> 
> Thoughts?

This is just a first pass to shuffle thing around into a directory. I plan to break things up a little more. As far as having a totally empty `__init__.py`  I don't have a strong preference either way.

**Comment by dwoz** [comment]
This work has been moved to #60595

**Pull request closed without merging**